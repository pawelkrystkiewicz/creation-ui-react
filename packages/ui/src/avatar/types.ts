import type React from 'react'
import { ReactNode } from 'react'

export const AVATAR_VARIANTS = ['circle', 'rounded', 'square'] as const

export type AvatarVariant = (typeof AVATAR_VARIANTS)[number]

type AvatarProps = Omit<React.ComponentProps<'img'>, 'size'> & {
  /**
   * How large should the Avatar be?
   */
  size?: number
  /**
   * Class name
   */
  className?: string
  /**
   * Variant of Avatar
   */
  variant?: AvatarVariant
  /**
   * Children of Avatar
   */
  children?: ReactNode
}

export default AvatarProps

export interface AvatarGroupProps
  extends Pick<AvatarProps, 'size' | 'variant'> {
  /**
   * Total number of avatars to show in counter
   */
  total?: number
  /**
   * Maximum number of avatars to show
   */
  limit?: number
  /**
   * Children of AvatarGroup - other `Avatar`s
   */
  children?: ReactNode
  /**
   * Controls the offset of `Avatar`s components.
   * Default -3.5% which translates directly to margin-left:-3.5%.
   */
  stackingOffsetPercent?: number
  /**
   * Class name
   */
  className?: string
  /**
   * `className` for surplus `Avatar` component.
   */
  surplusClassName?: string
  /**
   * Custom render function for surplus `Avatar` component.
   * @param surplusCount
   * @returns ReactNode
   */
  renderSurplus?: (surplusCount: number) => ReactNode
}
