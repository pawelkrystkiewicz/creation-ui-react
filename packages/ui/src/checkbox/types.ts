import * as Headless from '@headlessui/react'
import type { ReactNode } from 'react'

export interface CheckboxProps
  extends Omit<Headless.CheckboxProps, 'as' | 'className'> {
  children?: ReactNode
  className?: string
  readOnly?: boolean
  loading?: boolean
}
