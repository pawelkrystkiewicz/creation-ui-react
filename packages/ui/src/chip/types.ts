import { ReactNode } from 'react'
import type {
  BaseComponentProps,
  ElementBaseVariant,
  ElementColor,
  InputClassNamesAPI,
} from '../types'

export type ChipProps = Partial<Pick<BaseComponentProps, 'label'>> & {
  /**
   * Should be uppercase?
   */
  uppercase?: boolean
  /**
   * Status of the element
   */
  color?: ElementColor | 'unstyled'
  /**
   * Should have border?
   */
  border?: boolean
  /**
   * Variant
   */
  variant?: ElementBaseVariant
  /**
   * Start adornment
   */
  startAdornment?: ReactNode
  /**
   * Start adornment
   */
  endAdornment?: ReactNode
  /**
   * Callback when delete button is clicked
   */
  onDelete?: () => void
  /**
   *
   * Callback when chip is clicked
   */
  onClick?: () => void
  /**
   * Class names overrides
   */
  cx?: Pick<InputClassNamesAPI, 'container'>
  /**
   * Style overrides, helpful with adding custom colors. Passing `style` will disable `color` prop.
   */
  style?: React.CSSProperties
}
