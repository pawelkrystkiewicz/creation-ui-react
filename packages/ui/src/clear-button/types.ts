import type { ReactNode } from 'react'

export interface ClearButtonProps {
  children?: ReactNode
  className?: string
}
