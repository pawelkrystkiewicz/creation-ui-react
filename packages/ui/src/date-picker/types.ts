import { CalendarDateValue, CalendarProps } from '../calendar'
import { InputProps } from '../input'

export type DatePickerProps = Omit<CalendarProps, 'onChange'> &
  Omit<InputProps, 'onChange' | 'value' | 'ref'> & {
    onChange?: (date: CalendarDateValue) => void
    /**
     * z-index configuration
     */
    zIndex?: { popover?: number }
  }
