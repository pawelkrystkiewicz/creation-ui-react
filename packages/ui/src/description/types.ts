import type { ReactNode } from 'react'
import * as Headless from '@headlessui/react'


export interface DescriptionProps extends Omit<Headless.DescriptionProps, 'as' | 'className'> {
  children?: ReactNode
  className?: string
}
