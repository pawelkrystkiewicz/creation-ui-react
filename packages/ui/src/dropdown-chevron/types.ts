import { IconProps } from '../icon'

export interface DropdownChevronProps extends Omit<IconProps, 'icon'> {
  open?: boolean
}
