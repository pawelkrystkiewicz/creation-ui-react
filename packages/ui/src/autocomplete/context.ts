import { createContext, useContext } from 'react'
import {
  AutocompleteOptionProps,
  AutocompleteOptionDefault,
  AutocompleteProps,
  AutocompleteInnerInputProps,
} from './types'

export interface AutocompleteContextValue<T = any>
  extends Pick<
    AutocompleteProps<T>,
    | 'renderOption'
    | 'renderSelection'
    | 'getOptionLabel'
    | 'getLimitTagsText'
    | 'renderTags'
    | 'multiple'
    | 'clearable'
    | 'limit'
    | 'options'
    | 'textLoading'
    | 'textEmpty'
    | 'textNotFound'
    | 'defaultTagProps'
    | 'autoHighlight'
    | 'onCreate'
    | 'textCreate'
  > {
  query?: string
  allowCreate?: boolean
  open?: boolean
  floatingContext: any
  activeIndex: number | null
  selected?: T | T[] | null
  /** PROPS **/
  propsInput: AutocompleteInnerInputProps
  propsList: Record<string, unknown>
  getOptionProps: (option: T, index: number) => AutocompleteOptionProps
  /** CONTROLS **/
  setOpen: (value: boolean) => void
  handleRemoveSelected: (option: AutocompleteOptionDefault) => void
}

// Define a non-generic context first
// We will cast the type later
export const AutocompleteContext = createContext<any>(null)

export const useAutocomplete = <T = AutocompleteOptionDefault>() => {
  const context = useContext(
    AutocompleteContext,
  ) as AutocompleteContextValue<T> | null

  if (!context) {
    throw new Error(
      'useAutocomplete must be used within an AutocompleteProvider',
    )
  }
  return context
}
