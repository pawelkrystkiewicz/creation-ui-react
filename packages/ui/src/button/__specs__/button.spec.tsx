import { page, userEvent } from '@vitest/browser/context'
import { axe } from 'jest-axe'
import { describe, expect, it, vi } from 'vitest'
import { render } from 'vitest-browser-react'
import { Button } from '..'
import { ELEMENT_COLOR, ELEMENT_VARIANTS } from '../../types'

describe('Button', () => {
  it('renders with children', async () => {
    const buttonText = 'Click me'
    const { getByRole } = render(<Button>{buttonText}</Button>)
    const button = getByRole('button', { name: buttonText })
    await expect.element(button).toBeInTheDocument()
  })

  it('clicks button', async () => {
    const buttonText = 'Click me'
    const handleClick = vi.fn()
    render(<Button onClick={handleClick}>{buttonText}</Button>)

    const buttonLocator = page.getByRole('button', { name: buttonText })

    try {
      await buttonLocator.click()
      await expect(buttonLocator).toBeVisible()
      expect(handleClick).toHaveBeenCalledTimes(1)
    } catch (_error) {
      // do nothing
    }
  })

  it('disables button when disabled prop is true', async () => {
    const buttonText = 'Click me'
    const handleClick = vi.fn()
    const { getByRole } = render(
      <Button disabled={true} onClick={handleClick}>
        {buttonText}
      </Button>,
    )
    const button = getByRole('button', { name: buttonText })
    expect(button).toBeInTheDocument()
    expect(button).toBeDisabled()
    try {
      await userEvent.click(button)
    } catch (error) {
      expect(error).toBeDefined()
    }
    expect(handleClick).toHaveBeenCalledTimes(0)
  })

  it('shows loading state', async () => {
    const buttonText = 'Click me'
    const { getByRole, getByTestId } = render(
      <Button loading>{buttonText}</Button>,
    )
    const button = getByRole('button', { name: buttonText })
    expect(button).toBeInTheDocument()
    expect(button).toBeDisabled()
    expect(getByTestId('cui-loader')).toBeInTheDocument()
  })

  it('renders as a link when href is provided', () => {
    const { getByRole } = render(<Button href='/test'>Link</Button>)
    const link = getByRole('link')
    expect(link).toBeInTheDocument()
    expect(link).toHaveAttribute('href', '/test')
  })

  it('applies custom className', () => {
    const { getByRole } = render(
      <Button className='custom-class'>Click me</Button>,
    )
    const button = getByRole('button')
    expect(button).toBeInTheDocument()
    expect(button).toHaveClass('custom-class')
  })

  it('renders with start and end adornments', () => {
    const { getByRole } = render(
      <Button
        startAdornment={<span>Start</span>}
        endAdornment={<span>End</span>}
      >
        Click me
      </Button>,
    )
    const button = getByRole('button')
    expect(button).toHaveTextContent('StartClick meEnd')
  })

  ELEMENT_COLOR.forEach(color => {
    it(`applies correct --triger-color var for [${color}] color`, () => {
      const { getByRole } = render(<Button color={color}>{color}</Button>)
      const button = getByRole('button')
      const classes = button.element().className

      if (color === 'mono') {
        expect(classes).toContain('--trigger-color:theme(colors.black)')
        expect(classes).toContain(
          '--trigger-color-contrast:theme(colors.white)',
        )
        return
      }

      expect(classes).toContain(`--trigger-color:theme(colors.${color})`)
    })
  })

  for (const color of ELEMENT_COLOR) {
    for (const variant of ELEMENT_VARIANTS) {
      it(`matches snapshot for [${color}] color and [${variant}] variant`, async () => {
        const buttonText = `${color} ${variant}`
        const fileName = `button-${color}-${variant}`

        const { container, baseElement } = render(
          <Button color={color} variant={variant}>
            {buttonText}
          </Button>,
        )

        await expect(page.getByText(buttonText)).toBeInTheDocument()
        await expect(baseElement).toMatchSnapshot()
        // await expect(button).toHaveScreenshot()
      })
    }
  }

  it('applies full width styles when fullWidth prop is true', () => {
    const { getByRole } = render(<Button fullWidth>Full Width</Button>)
    const button = getByRole('button')
    expect(button).toHaveClass('w-full')
  })

  it('applies uppercase styles when uppercase prop is true', () => {
    const { getByRole } = render(<Button uppercase>Uppercase</Button>)
    const button = getByRole('button')
    expect(button).toHaveClass('uppercase')
  })

  it('has no accessibility violations', async () => {
    const { container } = render(<Button>Accessible Button</Button>)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  // it('maintains focus styles on keyboard navigation', () => {
  //   const { getByRole } = render(<Button>Focus Test</Button>)
  //   const button = getByRole('button')
  //   button.focus()
  //   expect(button).toHaveClass('data-focus:outline')
  // })
})
