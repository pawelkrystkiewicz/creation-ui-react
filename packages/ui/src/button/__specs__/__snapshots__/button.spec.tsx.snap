// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > matches snapshot for [error] color and [contained] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] text-[var(--trigger-color-contrast)] bg-[var(--trigger-color)] data-active:bg-[var(--trigger-color)]/[85%] data-hover:bg-[var(--trigger-color)]/90 [--trigger-color:theme(colors.error)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      error contained
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [error] color and [outlined] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border !border-[var(--trigger-color)] text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.error)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      error outlined
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [error] color and [text] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border-transparent text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.error)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      error text
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [mono] color and [contained] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] text-[var(--trigger-color-contrast)] bg-[var(--trigger-color)] data-active:bg-[var(--trigger-color)]/[85%] data-hover:bg-[var(--trigger-color)]/90 [--trigger-color:theme(colors.black)] [--trigger-color-contrast:theme(colors.white)] dark:[--trigger-color:theme(colors.white)] dark:[--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      mono contained
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [mono] color and [outlined] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border !border-[var(--trigger-color)] text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.black)] [--trigger-color-contrast:theme(colors.white)] dark:[--trigger-color:theme(colors.white)] dark:[--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      mono outlined
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [mono] color and [text] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border-transparent text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.black)] [--trigger-color-contrast:theme(colors.white)] dark:[--trigger-color:theme(colors.white)] dark:[--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      mono text
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [primary] color and [contained] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] text-[var(--trigger-color-contrast)] bg-[var(--trigger-color)] data-active:bg-[var(--trigger-color)]/[85%] data-hover:bg-[var(--trigger-color)]/90 [--trigger-color:theme(colors.primary)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      primary contained
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [primary] color and [outlined] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border !border-[var(--trigger-color)] text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.primary)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      primary outlined
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [primary] color and [text] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border-transparent text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.primary)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      primary text
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [success] color and [contained] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] text-[var(--trigger-color-contrast)] bg-[var(--trigger-color)] data-active:bg-[var(--trigger-color)]/[85%] data-hover:bg-[var(--trigger-color)]/90 [--trigger-color:theme(colors.success)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      success contained
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [success] color and [outlined] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border !border-[var(--trigger-color)] text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.success)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      success outlined
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [success] color and [text] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border-transparent text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.success)] [--trigger-color-contrast:theme(colors.white)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      success text
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [warning] color and [contained] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] text-[var(--trigger-color-contrast)] bg-[var(--trigger-color)] data-active:bg-[var(--trigger-color)]/[85%] data-hover:bg-[var(--trigger-color)]/90 [--trigger-color:theme(colors.warning)] [--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      warning contained
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [warning] color and [outlined] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border !border-[var(--trigger-color)] text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.warning)] [--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      warning outlined
    </button>
  </div>
</body>
`;

exports[`Button > matches snapshot for [warning] color and [text] variant 1`] = `
<body>
  
    
  
  


  <div>
    <button
      class="relative micro-interactions isolate inline-flex items-center justify-center gap-x-2 rounded-md font-medium px-[calc(theme(spacing[3.5])-1px)] py-[calc(theme(spacing[2.5])-1px)] sm:px-[calc(theme(spacing.3)-1px)] sm:py-[calc(theme(spacing[1.5])-1px)] focus:outline-none data-focus:outline data-focus:outline-2 data-focus:outline-offset-2 data-focus:outline-primary data-disabled:opacity-50 data-disabled:pointer-events-none data-active:scale-95 h-[var(--ui-height)] border-transparent text-[var(--trigger-color)] bg-transparent data-active:bg-[var(--trigger-color)]/20 data-hover:bg-[var(--trigger-color)]/10 dark:data-active:bg-[var(--trigger-color)]/30 dark:data-hover:bg-[var(--trigger-color)]/25 [--trigger-color:theme(colors.warning)] [--trigger-color-contrast:theme(colors.black)]"
      data-headlessui-state=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="absolute left-1/2 top-1/2 size-[max(100%,2.75rem)] -translate-x-1/2 -translate-y-1/2 [@media(pointer:fine)]:hidden cursor-pointer"
      />
      warning text
    </button>
  </div>
</body>
`;
